pkg/security/api/v1/types.go

```diff
@@ -80,16 +80,17 @@ type PodSecurityPolicyReview struct {
 // PodSecurityPolicyReviewSpec defines specification for PodSecurityPolicyReview
 type PodSecurityPolicyReviewSpec struct {
 	// template is the PodTemplateSpec to check. The template.spec.serviceAccountName field is used
-	// if serviceAccountNames is empty, unless the template.spec.serviceAccountName is empty,
+	// if serviceAccountNames is not nil and empty, unless the template.spec.serviceAccountName is empty,
 	// in which case "default" is used.
-	// If serviceAccountNames is specified, template.spec.serviceAccountName is ignored.
+	// If serviceAccountNames is specified or nil, template.spec.serviceAccountName is ignored.
```

Comment:
so the current behavior is
```
"foo", []    = ["foo"]
"foo", nil   = ["foo"]
"",    []    = ["default"]
"",    nil   = ["default"]
"foo", ["a"] = ["a"]
"",    ["a"] = ["a"]
```

and this is proposing changing these cases:
```
"foo", nil = ["default","builder","deployer",...]
"",    nil = ["default","builder","deployer",...]
```

that seems like surprising behavior to me